Src/Base/AMReX_GpuAtomic.H

```diff
@@ -22,13 +22,13 @@ namespace detail {
 
     template <typename R, typename I, typename F>
     AMREX_GPU_DEVICE AMREX_FORCE_INLINE
-    R atomic_op (R* address, R val, F f) noexcept
+    R atomic_op (R* AMREX_RESTRICT address, R const val, F const f) noexcept
```

Comment:
`const` in `R const` is certainly superfluous.

Reply:
are you sure? what's the type of `R`?

Reply:
Then what's `R*`?

Reply:
usually a `double*` in DP